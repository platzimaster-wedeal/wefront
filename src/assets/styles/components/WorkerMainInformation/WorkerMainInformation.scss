@import '../../partials/sizes.scss';
@import '../../partials/colors.scss';
@import '../../Breakpoints.scss';

.worker-main-information {
  position: relative;
  max-width: 320px;
  width: auto;
  padding: $space-size-big;


  @include media(tablet){
    max-width: 40%;
  }

  &__title {
    margin: $space-size-extra-small 0;
    font-weight: 400;
    @include font-size($font-size-normal);

    @include media(tablet) {
      @include font-size($font-size-normal);
    }
    
  }
}
