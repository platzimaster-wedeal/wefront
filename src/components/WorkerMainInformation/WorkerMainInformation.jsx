import React from "react";
import { MdShowChart, MdLanguage, MdStar } from "react-icons/md";

// Styles
import "../../assets/styles/components/WorkerMainInformation/WorkerMainInformation.scss";

import GraphicData from "../GraphicData/GraphicData";

const WorkerMainInformation = ({
 recommandantions = 82,
 languages = "Spanish",
 valoration = "4.5",
}) => (
 <div className="worker-main-information">
  <strong className="worker-main-information__title">Main Information</strong>
  <ul>
   <li>
    <GraphicData active={true} value={`${recommandantions} recommendations`}>
     <MdShowChart size={30} />
    </GraphicData>
   </li>
   <li>
    <GraphicData active={true} value={languages}>
     <MdLanguage size={30} />
    </GraphicData>
   </li>
   <li>
    <GraphicData active={true} value={`${valoration}/5`}>
     <MdStar size={30} />
    </GraphicData>
   </li>
  </ul>
 </div>
);

export default WorkerMainInformation;
