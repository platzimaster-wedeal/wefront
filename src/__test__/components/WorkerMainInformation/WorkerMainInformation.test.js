import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import WorkerMainInformation from "../../../components/WorkerMainInformation/WorkerMainInformation";

describe("Test for the component WorkerMainInformation :D", () => {
 const WorkerMainInformationMounted = mount(<WorkerMainInformation />);
 const WorkerMainInformationCreated = create(<WorkerMainInformation />);

 describe("Visual tests for the component", () => {
  test("should the comopnent have the same tree", () => {
   expect(WorkerMainInformationCreated.toJSON()).toMatchSnapshot();
  });
 });

 describe("Functional tests for the component", () => {
  test("Should the component render one element", () => {
   expect(WorkerMainInformationMounted.length).toEqual(1);
  });
  test("Should the title of the component be a string", () => {
   expect(
    typeof WorkerMainInformationMounted.find(
     ".worker-main-information__title"
    ).text()
   ).toEqual("string");
  });
  test("Should the title of the component be Worker Information", () => {
   expect(
    WorkerMainInformationMounted.find(".worker-main-information__title").text()
   ).toEqual("Main Information");
  });
 });
});
