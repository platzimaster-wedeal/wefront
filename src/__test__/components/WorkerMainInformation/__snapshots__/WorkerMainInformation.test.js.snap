// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the component WorkerMainInformation :D Visual tests for the component should the comopnent have the same tree 1`] = `
<div
  className="worker-main-information"
>
  <strong
    className="worker-main-information__title"
  >
    Main Information
  </strong>
  <ul>
    <li>
      <div
        className="graphic-data graphic-data--active"
      >
        <svg
          fill="currentColor"
          height={30}
          size={30}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width={30}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.5 18.49l6-6.01 4 4L22 6.92l-1.41-1.41-7.09 7.97-4-4L2 16.99z"
          />
        </svg>
        <strong>
          82 recommendations
        </strong>
      </div>
    </li>
    <li>
      <div
        className="graphic-data graphic-data--active"
      >
        <svg
          fill="currentColor"
          height={30}
          size={30}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width={30}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
          />
        </svg>
        <strong>
          Spanish
        </strong>
      </div>
    </li>
    <li>
      <div
        className="graphic-data graphic-data--active"
      >
        <svg
          fill="currentColor"
          height={30}
          size={30}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width={30}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
        <strong>
          4.5/5
        </strong>
      </div>
    </li>
  </ul>
</div>
`;
